jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -224,7 +227,7 @@ public boolean tryExecute(Runnable task)
             return false;
         }
 
-        int size = _size.decrementAndGet();
+        int size = _count.addAndGetLo(-1);
```

Comment:
This can make the `size` counter go negative. When this happens, the reserved threads will stop executing any task further on as `ReservedThread.run()` has the following check:

```
if (size >= 0 && size < _capacity)
    size++;
else
    exit = true;
```

so there's no way to make the counter go positive ever again and any value < 0 is interpreted as "the pool is stopping".